### PR TITLE
[Serializer] Docs for the PropertyInfo integration

### PR DESCRIPTION
**Pull request by dunglas**
Docs for symfony/symfony#17660.

### CODE REVIEW

File: components/serializer.rst [hunk 1]
```diff
@@ -671,6 +671,74 @@ you indicate that you're expecting an array instead of a single object.
...
+-----------------------------------------
+
+The Serializer Component can use the :doc:`PropertyInfo Component </components/property_info>` to denormalize
+complex types (objects). The type of the class's property will be guessed using the provided
+extractor and used to recursively denormalize the inner data.
+
+When using the Symfony Standard Edition, all normalizers are automatically configured to use registered extractors.
+When using the component standalone, an implementation of :class:`Symfony\\Component\\PropertyInfo\\PropertyTypeExtractorInterface`,
+(usually an instance of :class:`Symfony\\Component\\PropertyInfo\\PropertyInfoExtractor`) must be passed as the 4th
+parameter of the normalizer::
```
Comment (xabbuh) [inline components/serializer.rst]: "[...] to the `ObjectNormalizer`"?

File: components/serializer.rst [hunk 2]
```diff
@@ -671,6 +671,74 @@ you indicate that you're expecting an array instead of a single object.
...
     $persons = $serializer->deserialize($data, 'Acme\Person[]', 'json');
 
+Recursive Denormalization and Type Safety
+-----------------------------------------
+
+The Serializer Component can use the :doc:`PropertyInfo Component </components/property_info>` to denormalize
+complex types (objects). The type of the class's property will be guessed using the provided
+extractor and used to recursively denormalize the inner data.
+
+When using the Symfony Standard Edition, all normalizers are automatically configured to use registered extractors.
```
Comment (xabbuh) [inline components/serializer.rst]: [...] to use the registered [...]

File: components/serializer.rst [hunk 3]
```diff
@@ -671,6 +671,74 @@ you indicate that you're expecting an array instead of a single object.
     $data = ...; // The serialized data from the previous example
     $persons = $serializer->deserialize($data, 'Acme\Person[]', 'json');
 
+Recursive Denormalization and Type Safety
+-----------------------------------------
+
+The Serializer Component can use the :doc:`PropertyInfo Component </components/property_info>` to denormalize
+complex types (objects). The type of the class's property will be guessed using the provided
```
Comment (xabbuh) [inline components/serializer.rst]: class'

File: components/serializer.rst [hunk 4]
```diff
@@ -671,6 +671,74 @@ you indicate that you're expecting an array instead of a single object.
...
+         'Acme\ObjectOuter'
+    );
+
+    dump($obj->getInner()->foo); // 'foo'
+    dump($obj->getInner()->bar); // 'bar'
+    dump($obj->getDate()->format('Y-m-d')); // '1988-01-21'
+
+When a ``PropertyTypeExtractor`` is available, the normalizer will also check that the data to denormalize
+matches the type of the property (even for primitive types). For instance, if a ``string`` is provided but
+the type of the property is ``int``, a :class:`Symfony\\Component\\Serializer\\Exception\\UnexpectedValueException`
```
Comment (xabbuh) [inline components/serializer.rst]: an

File: components/serializer.rst (hunk 3 above)
Comment (dunglas) [inline components/serializer.rst]: Are you sure? http://grammar.ccc.commnet.edu/grammar/possessives.htm

File: components/serializer.rst [hunk 5]
```diff
@@ -671,6 +671,73 @@ you indicate that you're expecting an array instead of a single object.
...
+    }
+
+    class ObjectInner
+    {
+        public $foo;
+        public $bar;
+    }
+
+    $normalizer = new ObjectNormalizer(null, null, null, new ReflectionExtractor()); //
+    $serializer = new Serializer(array(new DateTimeNormalizer(), $normalizer));
```
Comment (dunglas) [inline components/serializer.rst]: done

**Comment by dunglas** [comment]
Rebased and fixed.

**Comment by xabbuh** [comment]
:+1: LGTM

Status: Reviewed

**Comment by xabbuh** [comment]
Thank you @dunglas.

**Pull request closed without merging**